lib/puppet/parser/resource.rb

```diff
@@ -304,6 +305,17 @@ def add_scope_tags
     end
   end
 
+  def replace_sensitive_data
+    if builtin_type?
+      parameters.each_pair do |name, param|
```

Comment:
There is a better way I think. (Will make separate comment about that). Problem with unwrapping on a `set` is that puppet logic can access the unwrapped value once it has been set.